var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -326,7 +326,7 @@ class Openmpi(AutotoolsPackage):
     # knem support was added in 1.5
     conflicts('fabrics=knem', when='@:1.4')
 
-    conflicts('schedulers=slurm ~pmi', when='@1.5.4:',
+    conflicts('schedulers=slurm ~pmi ~legacylaunchers', when='@1.5.4:',
```

Comment:
I don't understand the restriction here or configuration change below. The `legacylaunchers` option is just to remove the installed mpiexec/mpirun , not change the config in any other way.

Reply:
> If you're using legacylaunchers under slurm, as we are, you don't need pmi, and the slurm installation doesn't have development files it was built against. Without that change installation will fail looking for the library/header.

OK, I'm getting hung up  conflicts statement is a little confusing, and the `msg` didn't get updated for your change. I think this conflict statment better  conveys the intent:
```python
conflicts('~pmi ~legacylaunchers', when='@1.5.4: schedulers=slurm',
    msg='+pmi or '+legacylaunchers' is required for openmpi(>=1.5.5) to work with SLURM.')
```